requirements.txt

```diff
@@ -7,3 +7,4 @@ setuptools>=40.1.0
 typing_extensions
 h5py
 rustworkx
+sparse
```

Comment:
I know that in Qiskit Machine Learning this is handled as an optional dependency. Do we really need to do the same here? @woodsp-ibm @manoelmarques

Reply:
The underlying reason that prompted this being made optional in ML is described in Qiskit/qiskit-machine-learning#67 where at the time users wanted to use a platform where numba, a dependency of sparse, was not supported. It makes the source a bit more complex to cater to this being optional - you can see what was done in ML in the PR that resolved the issue. If the impact is not too bad we may want to consider optionality to have flexibility though still in the near term.

Reply:
Hm I see. I'm not sure whether Nature will ever be used on a platform like a Raspberry Pi. But I guess we should try to avoid making the `qiskit` metapackage dependent on numba.

Reply:
The platform in question is now supported by numba. You can see how the sparse optionality is handled in ML and what a similar approach might entail.

Reply:
If the platform is now supported, is there any other reason for making `sparse` an optional dependency? I would like to avoid making `sparse` optional since it will (presumably) be more and more useful to us when ironing out bottlenecks in the future.
If however there is a very good reason to make it optional, I guess we need to add the same `optionals.HAS_SPARSE` setting in Nature as we have in ML.

Reply:
It did not start out as optional in ML but was changed since it would fail to install where numba did not support the platform even if you had no intent/need of using sparse. If the impact seems low to the code it might be worth considering having it optional - e.g. Terra has just made tweedledum optional due to platform issues where it was not before since it ended up preventing installation/use.